scripts/missions/amk/06_An_Errand_The_Professors_Price.lua

```diff
@@ -20,28 +25,36 @@ mission.reward =
     nextMission = { xi.mission.log_id.AMK, xi.mission.id.amk.SHOCK_ARRANT_ABUSE_OF_AUTHORITY },
 }
 
+-- TODO: Test and make sure this works
 local orbKeyItems =
 {
-    xi.ki.ORB_OF_SWORDS,
-    xi.ki.ORB_OF_CUPS,
-    xi.ki.ORB_OF_BATONS,
-    xi.ki.ORB_OF_COINS,
+    -- keyItem, mod, immune value, vulnerable value
+    { xi.ki.ORB_OF_SWORDS, xi.mod.SLASH_SDT,       0, 1000 },
+    { xi.ki.ORB_OF_CUPS,   xi.mod.BLUNT_SDT,       0, 1000 },
+    { xi.ki.ORB_OF_BATONS, xi.mod.PIERCE_SDT,      0, 1000 },
+    { xi.ki.ORB_OF_COINS,  xi.mod.UDMGMAGIC,  -10000,    0 },
 }
 
 local beginCardianFight = function(player, npc)
     local numToSpawn = 15
 
-    -- TODO: Add immunities to cardians
+    local modsToAdd = {}
 
-    -- Remove KIs
+    -- Count KI's so we know how many Cardians to spawn
     local removedKIs = 0
-    for _, keyItemId in ipairs(orbKeyItems) do
+    for _, entry in ipairs(orbKeyItems) do
+        local keyItemId   = entry[1]
```

Comment:
I wonder, could this be simplified to `local keyItemId, immunity, immuneValue, vulnValue = unpack(entry)`?

Reply:
Yes, however I believe xi.mod.BLUNT_SDT needs to be replaced with IMPACT and HTH, so the table itself needs to be reworked.  I think @Xaver-DaRed is working on this so I'll leave it alone.